apps/native-component-list/src/screens/Clipboard/ClipboardScreen.tsx

```diff
@@ -0,0 +1,212 @@
+import * as Clipboard from 'expo-clipboard';
+import * as ImagePicker from 'expo-image-picker';
+import React from 'react';
+import { Text, ScrollView, StyleSheet, Alert } from 'react-native';
+
+import FunctionDemo, { FunctionDescription } from '../../components/FunctionDemo';
+import { ActionFunction, Platform } from '../../components/FunctionDemo/index.types';
+import { isCurrentPlatformSupported } from '../../components/FunctionDemo/utils';
+import ClipboardListenerDemo from './ClipboardListenerDemo';
+import ImagePreview from './ImagePreview';
+
+const withSupportedPlatforms = (platforms: Platform[], action: ActionFunction): ActionFunction =>
+  isCurrentPlatformSupported(platforms)
+    ? action
+    : () => Promise.resolve('Function is not supported on this platform');
+
+const HAS_X_ASYNC_CONFIG: FunctionDescription = {
+  name: 'hasXAsync',
```

Comment:
I'm not sure about this `X`, it's a little confusing, but I don't really have any better idea 👅 
`has<type>Async`, `has{String, Image, Url}Async`, `has{X}Async`? 🤔

Reply:
`hasSomethingAsync`?

Reply:
What if we split this config into three parts, separate for each method?

Reply:
> What if we split this config into three parts, separate for each method?

https://github.com/expo/expo/pull/16594#pullrequestreview-907167577